### PR TITLE
Refactor/check 358 reset change password pages

### PR DESCRIPTION
**Pull request by melsawy**
- Apply same design for login page to reset/change passwords 
-  Modernize `ChangePasswordComponent.js`, `UserPasswordChange.js` and `UserPasswordReset.js` components
- Define global text for reusing 
- Refactor inline styles

### CODE REVIEW

File: src/app/components/UserPasswordReset.js [hunk 1]
```diff
@@ -1,146 +1,209 @@
...
-                  defaultMessage="We've sent you an email from {adminEmail} with instructions to reset your password. Make sure it didn't wind up in your spam mailbox. If you aren't receiving our password reset emails, contact {supportEmail}. Please note that the link in this email will expire in {expiry} hours."
-                  values={{
-                    adminEmail: stringHelper('ADMIN_EMAIL'),
-                    supportEmail: stringHelper('SUPPORT_EMAIL'),
-                    expiry: Math.floor(this.state.expiry / 3600),
-                  }}
+  const classes = useStyles();
+
+  return (
+    <PageTitle>
```
Comment (amoedoamorim) [inline src/app/components/UserPasswordReset.js:129]: PageTitle isn't setting anything

File: src/app/components/UserPasswordReset.js [hunk 2]
```diff
@@ -1,146 +1,209 @@
...
-    );
-  }
-}
+              </div>
+            </CardContent>,
+            <CardActions key="usr-3" className="user-password-reset__actions">
+              <Button onClick={handleGoBack}>
+                <FormattedMessage {...globalStrings.cancel} />
+              </Button>
+              <Button color="primary" disabled={submitDisabled} onClick={(e) => { handleSubmit(e); }}>
```
Comment (amoedoamorim) [inline src/app/components/UserPasswordReset.js:191]: `onClick={handleSubmit}`

File: src/app/components/UserPasswordReset.js [hunk 3]
```diff
@@ -1,146 +1,209 @@
...
+          <CheckAgreeTerms />
+        </Box>
+      </ContentColumn>
+    </PageTitle>
+  );
+};
 
 UserPasswordReset.contextTypes = {
   store: PropTypes.object,
 };
```
Comment (amoedoamorim) [inline src/app/components/UserPasswordReset.js:208]: Please specify component propTypes

File: src/app/styles/js/shared.js [hunk 4]
```diff
@@ -389,6 +348,21 @@ export const HeaderTitle = styled.h3`
   `}
 `;
 
+export const StyledSubHeader = styled.h2`
+  font: ${title1};
+  font-weight: 600;
+  color: ${black54};
+  text-align: center;
+  margin-top: ${units(2)};
+`;
```
Comment (amoedoamorim) [inline src/app/styles/js/shared.js:357]: Fine for now but would be best to use `Typography` and/or a dedicated component under `components/layout`. Please add a FIXME comment

File: src/app/styles/js/shared.js [hunk 5]
```diff
@@ -389,6 +348,21 @@ export const HeaderTitle = styled.h3`
...
+
+export const StyledCard = styled(Card)`
+  padding: ${units(11)} ${units(15)} ${units(3)} !important;
+  ${mediaQuery.handheld`
+    padding: ${units(8)} ${units(4)} ${units(3)} !important;
+  `}
+`;
+
```
Comment (amoedoamorim) [inline src/app/styles/js/shared.js:365]: Fine for now but would be best to use a dedicated component under `components/layout`. Also naming should ideally state a clear purpose for the component. `StyledCard` is a bit vague. I'd suggest creating a `CheckBrandedCard` component.
Please add a FIXME comment.

File: src/app/components/source/UserSecurity.js [hunk 6]
```diff
@@ -492,17 +497,17 @@ class UserSecurity extends Component {
         <h2 style={style}>
           <FormattedMessage id="userSecurity.changePassword" defaultMessage="Change password" />
         </h2>
-        <StyledPasswordChange>
-          <Card style={style} className="user-password-change__card">
+        <ContentColumn center className="user-password-reset__component">
+          <StyledCard>
```
Comment (amoedoamorim) [inline src/app/components/source/UserSecurity.js:501]: I guess a regular `Card` will do here, no need for the extra top padding added by `StyledCard`.

File: src/app/components/UserPasswordChange.js [hunk 7]
```diff
@@ -1,81 +1,111 @@
...
-                </Card>
-              }
-            </StyledPasswordChange>
-          </PageTitle>
-        )}
-      </FormattedMessage>
-    );
-  }
+          { showConfirmDialog ?
+            <div>
```
Comment (amoedoamorim) [inline src/app/components/UserPasswordChange.js:78]: Grouping divs can be replaced with `React.Fragment`

File: src/app/components/LoginContainer.js [hunk 8]
```diff
@@ -1,24 +1,35 @@
...
 
 const LoginContainer = props => (
   <Typography component="div" variant="body2" gutterBottom>
     <PageTitle>
       <ContentColumn center id="login-container" className="login-container">
         <Favicon url={`/images/logo/${config.appName}.ico`} animated={false} />
 
-        <p style={{ marginTop: 16, textAlign: 'center' }}>
-          <FormattedHTMLMessage id="browser.support.message" defaultMessage='Best viewed with <a href="https://www.google.com/chrome/browser/desktop/">Chrome for Desktop</a>.' />
+        <p style={styles.bestViewed}>
```
Comment (amoedoamorim) [inline src/app/components/LoginContainer.js:31]: I'd rather use `Box` for layouting

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Looking great, Sawy! Great job! Lots of improvements here! I've added a few notes to help improve it further!

File: src/app/components/ChangePasswordComponent.js [hunk 9]
```diff
@@ -1,154 +1,173 @@
...
-              onChange={this.handleChangePasswordConfirm.bind(this)}
+  return (
+    <div className="user-password-change__password-input">
+      <Message message={errorMsg} />
+      {showCurrentPassword === true ? (
+        <TextField
+          className="user-password-change__password-input-field"
+          id="password-change-password-input-current"
+          type="password"
+          onChange={(e) => { handleChangeCurrentPassword(e); }}
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js:121]: But here I'd like to catch `e.target.value` in `handleChangeCurrentPassword` how can I do this when passing `handleChangeCurrentPassword` without argument?

File: src/app/components/ChangePasswordComponent.js [hunk 10]
```diff
@@ -1,154 +1,173 @@
...
-import ChangePasswordMutation from '../relay/mutations/ChangePasswordMutation';
 import { getErrorMessage } from '../helpers';
+import { units } from '../styles/js/shared';
 
-// TODO Read this from the backend.
-const passwordLength = {
-  min: 8,
-  max: 128,
-};
+const useStyles = makeStyles({
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js]: Done

File: src/app/components/ChangePasswordComponent.js [hunk 11]
```diff
@@ -1,154 +1,173 @@
...
+  marginTop: {
+    marginTop: `${units(3)}`,
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js]: Done

File: src/app/components/ChangePasswordComponent.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js:120]: Done

File: src/app/components/ChangePasswordComponent.js [hunk 12]
```diff
@@ -1,154 +1,173 @@
...
-        </FormattedMessage>
-        <br />
+          }
+        />
+      ) : null}
+      <TextField
+        className="user-password-change__password-input-field"
+        id="password-change-password-input"
+        type="password"
+        onChange={(e) => { handleChangePassword(e); }}
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js]: Done

File: src/app/components/ChangePasswordComponent.js [hunk 13]
```diff
@@ -1,154 +1,173 @@
...
+            defaultMessage="New password (minimum {min} characters)"
+            values={{ min: passwordLength.min }}
+          />
+        }
+      />
+      <TextField
+        className="user-password-change__password-input-field"
+        id="password-change-password-input-confirm"
+        type="password"
+        onChange={(e) => { handleChangePasswordConfirm(e); }}
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/ChangePasswordComponent.js]: Done

File: src/app/styles/js/shared.js (hunk 4 above)
Comment (melsawy) [reply src/app/styles/js/shared.js:358]: Added FIXME comment

File: src/app/styles/js/shared.js (hunk 5 above)
Comment (melsawy) [reply src/app/styles/js/shared.js:367]: Added FIXME comment

File: src/app/components/source/UserSecurity.js (hunk 6 above)
Comment (melsawy) [reply src/app/components/source/UserSecurity.js]: Replaced `StyledCard` with `Card`

File: src/app/components/UserPasswordChange.js (hunk 7 above)
Comment (melsawy) [reply src/app/components/UserPasswordChange.js]: Done

File: src/app/components/LoginContainer.js (hunk 8 above)
Comment (melsawy) [reply src/app/components/LoginContainer.js]: Done and used `Box` for layouting

**Pull request merged**